### PR TITLE
[Avatar / Presence Badge] Android changes / remove awayOutOfOffice / Bug fixes

### PR DESCRIPTION
**Pull request by ayush547**
### Platforms Impacted
- [x] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [x] android

### Description of changes

- Fixed 'away' 'outOfOffice' combined state not getting detected in styling due to lower precedence, removed 'awayOutOfOffice' state and updated code to work with individual 'away' and 'outOfOffice' states.
- Android Avatar changes to include presence badge.
- Docs update for Avatar & Presence badge.
- Added badgeX, badgeY tokens for Avatar presence badge.
- Avatar and Presence Badge tokenization Android.
- Forked getFallbackIconPath to support filled and unfilled icons required for Android Avatar.
- Fixed Avatar size not being applied correctly for Android.
- Updated presence badge positioning token values for Android, based on design assumptions followed.
- Fixed color bug in accent avatar.
- Updated stylingUtils.ts size computation to fix size bug.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot_2023-02-08-19-57-35-32_3238b23fbbdef38ccf4a465a34706bab | Screenshot_2023-02-27-14-29-57-47_3238b23fbbdef38ccf4a465a34706bab |

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by harrieshin** [comment]
bit orthogonal to this PR but the example "pawn" style avatar isn't really based on fluent mobile design. These are the only variation we should support 
image

**Comment by harrieshin** [comment]
the presence ring on smaller avatar sizes doesn't seem to be drawing correctly?

**Comment by ayush547** [comment]
> bit orthogonal to this PR but the example "pawn" style avatar isn't really based on fluent mobile design. These are the only variation we should support image

Could you share a bit more details on what the "pawn" style refers to?

**Comment by ayush547** [comment]
> the presence ring on smaller avatar sizes doesn't seem to be drawing correctly?

Known issue, adding a fix

**Comment by ayush547** [comment]
> is the presense offset actually correct in the smaller avatar sizes? it looks different from the spec image

Converting PR to draft to pick this up with design offline

File: packages/components/Badge/SPEC.md [hunk 1]
```diff
@@ -267,6 +267,8 @@ Presence badge represents user's availability.
 
 ### PresenceBadge Props
```
Comment (ankraj12) [inline packages/components/Badge/SPEC.md:268]: Query unrelated to your change - is `PresenceBadge` exported as a component for it's used only within library for avatar?

File: packages/components/Avatar/src/stylingUtils.ts [hunk 2]
```diff
@@ -59,3 +57,9 @@ export function getRingSpacing(tokens: AvatarTokens) {
       }
     : {};
 }
+
+// Android has filled and unfilled icons for different Avatars, the avatarColor prop is used to determine which icon to use.
+// Not required for other platforms.
+export function getFallbackIconPath(_avatarColor: AvatarColor) {
```
Comment (ankraj12) [inline packages/components/Avatar/src/stylingUtils.ts:63]: You can make it an optional param.

File: packages/components/Avatar/SPEC.md [hunk 3]
```diff
@@ -203,6 +205,16 @@ export interface AvatarTokens extends IBackgroundColorTokens, IForegroundColorTo
    */
   badgeSize?: BadgeSize;
 
+  /**
+   * The x position of presence badge.
+   */
```
Comment (ankraj12) [inline packages/components/Avatar/SPEC.md:210]: Can you elaborate position relative to which element as [0,0].

File: packages/components/Avatar/src/Avatar.styling.ts [hunk 4]
```diff
@@ -187,9 +185,11 @@ export const stylingSettings: UseStylingOptions<AvatarProps, AvatarSlotProps, Av
           status: tokens.badgeStatus,
           position: 'absolute',
           outOfOffice: tokens.outOfOffice,
+          right: tokens.badgeX !== undefined ? tokens.badgeX : undefined,
```
Comment (ankraj12) [inline packages/components/Avatar/src/Avatar.styling.ts:188]: Which case has this? Is it size16 - what is happens if you don't check on this?

File: packages/components/Avatar/SPEC.md [hunk 5]
```diff
@@ -83,8 +83,10 @@ The Avatar supports a circular and square (with rounded corners) shape.
 
 ### Sizes
 
-`Avatar` supports next sizes: `20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 96, 120`.
-Default size for win32 is 24.
+`Avatar` supports the following sizes: `20, 24, 28, 32, 36, 40, 48, 56, 64, 72, 96, 120`.
+Android only supports the following sizes: `16, 20, 24, 32, 40, 56, 72`. Badge is not displayed for Size 16.
```
Comment (ankraj12) [inline packages/components/Avatar/SPEC.md:87]: nit: and AcitivtyRing

File: apps/fluent-tester/src/TestComponents/Avatar/BasicAvatar.android.tsx [hunk 6]
```diff
@@ -78,6 +100,17 @@ export const StandardUsage: FunctionComponent = () => {
         <Avatar active={activityRing ? 'active' : 'inactive'} activeAppearance="ring" size={72} avatarColor={'brand'} />
       </View>
 
+      <View style={mobileStyles.testVariants}>
+        {/* Variation - Standard */}
```
Comment (ankraj12) [inline apps/fluent-tester/src/TestComponents/Avatar/BasicAvatar.android.tsx:104]: nit: Standard _OutOfOffice_

File: packages/components/Avatar/src/useAvatar.ts [hunk 7]
```diff
@@ -76,6 +77,11 @@ export const useAvatar = (props: AvatarProps): AvatarInfo => {
     };
   }
 
+  const disableBadge = Platform.OS === 'android' && size === 16;
+  if (disableBadge) {
+    state.showBadge = false;
+  }
```
Comment (ankraj12) [inline packages/components/Avatar/src/useAvatar.ts:83]: How did you check for ring removal?

File: packages/components/Avatar/src/AvatarTokens.mobile.ts [hunk 8]
```diff
@@ -108,7 +120,7 @@ export const defaultAvatarTokens: TokenSettings<AvatarTokens, Theme> = (t: Theme
       ringColor: t.colors.brandStroke1,
     },
     accent: {
-      backgroundColor: t.colors.brandBackgroundTint,
+      backgroundColor: t.host.appearance === 'light' ? t.colors.brandBackgroundTint : t.colors.neutralBackground1,
```
Comment (ankraj12) [inline packages/components/Avatar/src/AvatarTokens.mobile.ts:123]: Thank you for fixing this, and other trading issues in variations :)

**Review by ankraj12 — APPROVED** [review]
Can you share a light mode snippet?
Also, consider making the "android" check towards mobile checks - it will help keeping code inline unless needing a change in iOS.

File: packages/components/Avatar/SPEC.md (hunk 5 above)
Comment (ayush547) [reply packages/components/Avatar/SPEC.md:87]: ActivityRing is displayed for size16, only presenceBadge is disabled.

File: apps/fluent-tester/src/TestComponents/Avatar/BasicAvatar.android.tsx (hunk 6 above)
Comment (ayush547) [reply apps/fluent-tester/src/TestComponents/Avatar/BasicAvatar.android.tsx:104]: Updating to "Standard with Badge"

File: packages/components/Avatar/src/stylingUtils.ts (hunk 2 above)
Comment (ayush547) [reply packages/components/Avatar/src/stylingUtils.ts:63]: Updated.

File: packages/components/Avatar/src/useAvatar.ts (hunk 7 above)
Comment (ayush547) [reply packages/components/Avatar/src/useAvatar.ts:83]: Ring removal not required as per design guidelines

File: packages/components/Avatar/SPEC.md (hunk 3 above)
Comment (ayush547) [reply packages/components/Avatar/SPEC.md:210]: Updated.

File: packages/components/Avatar/src/Avatar.styling.ts (hunk 4 above)
Comment (ayush547) [reply packages/components/Avatar/src/Avatar.styling.ts:188]: BadgeX and BadgeY, when present override the default badge position picked from presenceBadge tokens. This check prevents unnecessary overrides.
Added in comments.

**Pull request merged**